### PR TITLE
Fixes Autoposition in some cases

### LINKED ISSUE #681: Bug: Calculating auto position removes top/bottom placement options incorrectly
Easy to reproduce, just open the auto position demo page and shrink your window enough.

image

Caused by this line in `_determineAutoPosition` function.
```javascript
661:  if (targetOffset.left + tooltipWidth > windowSize.width || ((targetOffset.left + (targetOffset.width / 2)) - tooltipWidth) < 0) {
```
What is the purpose of the second part of the condition? I don't see any reason to have it there. Even if the tooltip had been positioned in something like `bottom-center` you would have to divide `tooltipWidth/2`.

I think this should be removed, it basically breaks the `auto` positioning on mobile devices.

### PR DESCRIPTION
**Pull request by bozdoz**
Removed the latter half of an if statement in _determineAutoposition, as recommended in #681.  

Added 'introjs-' position to the tooltip className so users can customize css by that if they wish.

### CODE REVIEW

**Comment by bozdoz** [comment]
With the scroll position included in autopositioning, the tooltip would appear on screen in situations where it might not have if the user scrolled:

Normal position
top-align

Autopositioning on the bottom, after scroll:
auto-bottom-align

**Comment by afshinm** [comment]
This is awesome @bozdoz! I'm going to test it locally.

**Comment by afshinm** [comment]
I noticed a few issues with this branch but looks good in general. I'm testing on Firefox.

2017-10-08-141940_1016x1671_scrot
2017-10-08-141933_1016x1671_scrot

**Comment by bozdoz** [comment]
I'll take a look

**Comment by bozdoz** [comment]
Something looks very wrong there @afshinm.  I wondered if maybe somehow there was something off with Firefox, but I'm testing on it now and can't reproduce your screenshots.  I notice that they aren't right-aligned, first of all (they don't have the arrow on the top-right).  And they're obviously positioned off the page.  I'm trying Firefox, on responsive design mode, with Google Nexus 5 selected (though I've tried a few others too).  I'm not sure what is happening.  Any suggestions, ideas?

**Comment by afshinm** [comment]
Hmm, I'm using i3 wm and I didn't test that on responsive mode, the window
width was not big enough to fit the page.

I will test it again on Google Chrome.

**Comment by bozdoz** [comment]
Did you display the hints and then shrink the page?

**Comment by bozdoz** [comment]
Nevermind, I reproduced the error by shrinking screen size, as opposed to using responsive mode.  I'll take another look now.

File: intro.js [hunk 1]
```diff
@@ -609,37 +609,15 @@
...
 
-    if (currentTooltipPosition != "floating") { // Floating is always valid, no point in calculating
-      if (currentTooltipPosition === "auto") {
-        currentTooltipPosition = _determineAutoPosition.call(this, targetElement, tooltipLayer);
-      } else {
-        currentTooltipPosition = _determineAutoPosition.call(this, targetElement, tooltipLayer, currentTooltipPosition);
-      }
+    // Floating is always valid, no point in calculating
+    if (currentTooltipPosition != "floating") { 
+      currentTooltipPosition = _determineAutoPosition.call(this, targetElement, tooltipLayer, currentTooltipPosition);
```
Comment (bozdoz) [inline intro.js]: _determineAutoPosition already checks for 'auto'

File: intro.js [hunk 2]
```diff
@@ -864,10 +853,66 @@
       }
     }
 
+    // only top and bottom positions have optional alignments
+    if (['top', 'bottom'].indexOf(calculatedPosition) !== -1) {
+      calculatedPosition += _determineAutoAlignment(targetOffset.left, tooltipWidth, windowSize, desiredAlignment);
```
Comment (bozdoz) [inline intro.js]: alignment is calculated after position, when we know whether the auto-position will be top or bottom

**Comment by bozdoz** [comment]
Testing is already proving its worth! :D

**Comment by gilesbutler** [comment]
Any idea when this will be merged and released guys? :)

**Comment by afshinm** [comment]
Looks good to me, feel free to merge @bozdoz

**Comment by gilesbutler** [comment]
Thanks for the update @afshinm and @bozdoz 🙂 do you have a rough eta on the next release at all?

File: intro.js [hunk 3]
```diff
@@ -757,40 +782,44 @@
...
-        _removeEntry(possiblePositions, "bottom");
-      }
+    }
 
-      // Check for space above
-      if (targetOffset.top - tooltipHeight < 0) {
-        _removeEntry(possiblePositions, "top");
-      }
+    // Check for space above
+    if (targetOffset.top - tooltipHeight < 0) {
```
Comment (tianyong90) [inline intro.js]: When the target-element is fixed at the bottom, just like many tabbar menus do, the expression will always result a true value, however, the space above is big enough……

**Pull request merged**